containers/agent/deploy.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	deployCmd = &cobra.Command{
+		Use:   "deploy",
+		Short: "Execute an OpenStack Deploy workflow",
+		Long:  "",
+		Run:   runDeployCmd,
+	}
+
+	deployOpts struct {
+		kubeconfig     string
+		namespace      string
+		pod            string
+		configVersion  string
+		gitURL         string
+		gitSSHIdentity string
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(deployCmd)
+	deployCmd.PersistentFlags().StringVar(&deployOpts.namespace, "namespace", "openstack", "Namespace to use for openstackclient pod.")
+	deployCmd.PersistentFlags().StringVar(&deployOpts.pod, "pod", "", "Pod to use for executing the deployment.")
+	deployCmd.PersistentFlags().StringVar(&deployOpts.configVersion, "configVersion", "", "Config version to use when executing the deployment.")
+	deployCmd.PersistentFlags().StringVar(&deployOpts.gitURL, "gitURL", "", "Git URL to use when downloading playbooks.")
+	deployCmd.PersistentFlags().StringVar(&deployOpts.gitSSHIdentity, "gitSSHIdentity", "", "Git SSH Identity to use when downloading playbooks.")
+}
+
+// ExecPodCommand -
+func ExecPodCommand(kclient kubernetes.Clientset, pod corev1.Pod, containerName string, command string) error {
+	req := kclient.CoreV1().RESTClient().Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		Param("container", containerName).
+		Param("stdin", "true").
+		Param("stdout", "true").
+		Param("stderr", "true").
+		Param("tty", "false").
+		Param("command", "sh")
+
+	cfg, err := config.GetConfig()
+
+	if err != nil {
+		return err
+	}
+
+	exec, err := remotecommand.NewSPDYExecutor(cfg, "POST", req.URL())
+	if err != nil {
+		return err
+	}
+
+	argString := fmt.Sprintf("%s\n", command)
+	reader := strings.NewReader(argString)
+
+	return exec.Stream(remotecommand.StreamOptions{
+		Stdin:  io.Reader(reader),
+		Stdout: os.Stdout,
```

Comment:
@abays I modified these to use osStdout and os.Stderr. That way we get all the data streamed back to the pod logs in real time instead of once at the end.